Source/Details/ASDataController.mm

```diff
@@ -437,6 +437,29 @@ - (void)waitUntilAllUpdatesAreCommitted
   [self _scheduleBlockOnMainSerialQueue:^{ }];
 }
 
+- (BOOL)isProcessingUpdates
+{
+  ASDisplayNodeAssertMainThread();
+  long waitResult = dispatch_group_wait(_editingTransactionGroup, DISPATCH_TIME_NOW);
+  return (waitResult == 0 ? NO : YES);
```

Comment:
Is it possible that we could return a false `NO` if they call this in between the time the editing transaction group is finished and the main queue services the data controller's UIKit flush?

We could either flush the main serial queue (but not the editing transaction queue) or add API to check if the main serial queue is empty.